### ISSUE TITLE
How to use aliased column?

### ISSUE THREAD

**Issue by mattking17** [JetBrains/Exposed#459]
I'm trying to find in the docs and the source how to define an aliased column. By that I mean something like the following:

`select concat(users.first_name, ' ', users.last_name) as full_name from users;`

The example is trivial, but the use case I have is not - my real life case is to use built in SQL functions to calculate distance of a database record's latitude/longitude from a given point, and return it as a column alias `distance`.

Does this exist in Exposed, and I'm just missing it? Or is there another way to accomplish what I need?

**Comment by bradcypert** [comment]
@mattking17 As far as Aliasing goes, here are the docs for using alias with the DSL - https://github.com/JetBrains/Exposed/wiki/DSL#alias

I'm not sure if that will get you close to what you're looking for but it may help :)

Also...

```
transaction {
  // print sql to std-out
  addLogger(StdOutSqlLogger)
} 
```

Can help you debug by logging the executed SQL. I've used that in the past to see what was running and make a few changes (And test each one) until I got exactly what I wanted from Exposed.

**Comment by mattking17** [comment]
Thanks for the reply. I'm not sure that's what I'm looking for, it aliases tables and columns, but how would I go about constructing SQL functions as an alias? For instance:

```
6371 * ACOS(COS(RADIANS(:latitude)) * COS(RADIANS(`lat`)) * COS(RADIANS(:longitude) - RADIANS(`lng`)) + SIN(RADIANS(:latitude)) * SIN(RADIANS(`lat`))) AS `distance`
```

Where `:latitude` and `:longitude` could be passed in as arguments, and `lat` and `lng` are real columns. It would be nice to define this somehow on a DAO object, and have it be fetched alongside the other defined columns when executing a query.

I thought perhaps it could be defined as a custom `Operator` or `Expression` from Exposed, but I wasn't able to figure out how I could put that together.

**Comment by bradcypert** [comment]
Oh, I misunderstood! I'm not sure how to go about exactly what you've mentioned above. I don't believe exposed has support for populating properties based off of the results of specific SQL functions.

If you're blocked by this, a potential workaround may be to write a function on the DAO object that does the math in Kotlin? Sorry if that's not a huge help.

**Comment by mattking17** [comment]
Yep, that's exactly what I did after I pulled the data out of the database and converted it the actual entity objects used in the application. Would be nice to do that in the query but it works for now.

**Comment by Tapac** [comment]
@mattking17 , look for this as for an example:
```kotlin
    class Cos(val expr: Expression<Int>): Function<Int>(IntegerColumnType()) {
        override fun toSQL(queryBuilder: QueryBuilder): String = "COS(${expr.toSQL(queryBuilder)})"
    }
    fun distance(lat: Int, long:Int) = Expression.build {
        Cos(intParam(lat)) * Cos(intParam(long))  - Cos(Points.longitude)
    }
    
     val distance = distance(1, 2).alias("distance")
     val distances = cities.slice(distance).selectAll().map { it[distance] }
```

**Comment by jeffnyauke** [comment]
> ```kotlin
>     class Cos(val expr: Expression<Int>): Function<Int>(IntegerColumnType()) {
>         override fun toSQL(queryBuilder: QueryBuilder): String = "COS(${expr.toSQL(queryBuilder)})"
>     }
>     fun distance(lat: Int, long:Int) = Expression.build {
>         Cos(intParam(lat)) * Cos(intParam(long))  - Cos(Points.longitude)
>     }
>     
> ```

I'm getting errors here. Does this work with the latest vesrion of exposed?

**Comment by Tapac** [comment]
@jeffnyauke , please file a new issue with the details
